python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -191,16 +300,59 @@ def extract_message_or_forward_to_stdout(handler: "PipesMessageHandler", log_lin
 )
 
 
+@experimental
 @contextmanager
 def open_pipes_session(
     context: OpExecutionContext,
     context_injector: PipesContextInjector,
     message_reader: PipesMessageReader,
     extras: Optional[PipesExtras] = None,
 ) -> Iterator[PipesSession]:
-    """Enter the context managed context injector and message reader that power the EXT protocol and receive the environment variables
-    that need to be provided to the external process.
+    """Context manager that opens and closes a pipes session.
+
+    This context manager should be used to wrap the launch of an external process using the pipe
+    protocol to report results back to Dagster. The yielded :py:class:`PipesSession` should be used
+    to (a) obtain the environment variables that need to be provided to the external process; (b)
+    access results streamed back from the external process.
+
+    This method can be considered a lower-level alternative to :py:class:`PipesClient` subclasses.
```

Comment:
"This method is an alternative to :py:class:`PipesClient` subclasses for users who want more control over how pipes processes are launch. When using `open_pipes_session`, it is the user's responsibility to access the environment variables and pass them to the appropriate API to launch the remote process."